.github/workflows/main.yml

```diff
@@ -1,14 +1,75 @@
-on: [push]
+name: Build and upload Mender Artifact
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+env:
+  MENDER_SERVER_URL: https://hosted.mender.io           # Mender server URL
+  MENDER_ARTIFACT_NAME: artifact.mender                 # Mender artifact name
+  MENDER_RELEASE_NAME: change-the-world-to-better-v1.0  # Mender release name
+  ARTIFACTS_LIST_FILE_NAME: artifacts_list.json
 
 jobs:
   upload:
     runs-on: ubuntu-latest
-    name: Upload a dummy Mender artifact
+    # conrainer with pre-installed mender-artifact and mender-cli tools
+    container:
+      image: mendersoftware/mender-ci-tools:mender-master
+    name: Upload a dummy Mender Artifact
     steps:
+      # Checkout git repository
       - uses: actions/checkout@v3
-      - name: push artifact to hosted Mender
+      - name: Setup environment
+        run: |
+          apk add jq
+          # secrets are not passed to the runner when a workflow is triggered from a forked repository
+          # https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow
+          if [[ -z "${{ secrets.MENDER_SERVER_ACCESS_TOKEN }}" || "${{ secrets.MENDER_SERVER_ACCESS_TOKEN }}" == "" ]]; then
```

Comment:
Same as in https://github.com/mendersoftware/mender-gh-action-create-deployment/pull/3#discussion_r1031159102